src/awesome_map/BoundaryInterceptor.js

```diff
@@ -373,6 +474,65 @@ define(function(require) {
         },
 
         /**
+         * Tests to see if the given event state meets the criteria for firing
+         * one or more onScrollPastBoundary events.
+         * @param {TransformState} targetState
+         * @private
+         */
+        _checkForBoundaryEvents: function(targetState) {
+            var boundedPosition = this._getBoundedPosition(targetState);
+            var deltaY = targetState.translateY - boundedPosition.y;
+            if (deltaY > this._boundarySensitivity &&
+                this._visibleBoundaries.indexOf(BoundaryTypes.TOP) !== -1) {
+                this.onScrollPastBoundary.dispatch([this, {
+                    boundary: BoundaryTypes.TOP
+                }]);
+            }
+            if (deltaY < -this._boundarySensitivity &&
+                this._visibleBoundaries.indexOf(BoundaryTypes.BOTTOM) !== -1) {
+                this.onScrollPastBoundary.dispatch([this, {
+                    boundary: BoundaryTypes.BOTTOM
+                }]);
+            }
+            var deltaX = targetState.translateX - boundedPosition.x;
+            if (deltaX > this._boundarySensitivity &&
+                this._visibleBoundaries.indexOf(BoundaryTypes.LEFT) !== -1) {
+                this.onScrollPastBoundary.dispatch([this, {
+                    boundary: BoundaryTypes.LEFT
+                }]);
+            }
+            if (deltaX < -this._boundarySensitivity &&
+                this._visibleBoundaries.indexOf(BoundaryTypes.RIGHT) !== -1) {
+                this.onScrollPastBoundary.dispatch([this, {
+                    boundary: BoundaryTypes.RIGHT
+                }]);
+            }
+        },
+
+        /**
+         * A debounced wrapper for _determineBoundaryVisibility.
+         * This is declared now for clarity, but overriden at the end of the Prototype
+         * definition because lodash requires a concrete function.
+         * @private
+         */
+        _debouncedDetermineBoundaryVisibility: function() {
+        },
+
+        /**
+         * Creates a new list of currently visible boundaries.
+         * @private
+         */
+        _determineBoundaryVisibility: function() {
+            this._visibleBoundaries = [];
+            // Determine boundary visibility
+            for (var boundary in BoundaryTypes) {
+                if (this.isBoundaryVisible(BoundaryTypes[boundary])) {
```

Comment:
Well it's not strictly necessary in either place, since the function itself filters the input, but jshint yelled at me for not having in the other loop.  I can add it here too so it's consistent